src/aof.c

```diff
@@ -2530,9 +2531,10 @@ void backgroundRewriteDoneHandler(int exitcode, int bysignal) {
 
         if (server.aof_fd != -1) {
             /* AOF enabled. */
+            off_t base_size = getAppendOnlyFileSize(new_base_filename, NULL);
             server.aof_selected_db = -1; /* Make sure SELECT is re-issued */
-            server.aof_current_size = getAppendOnlyFileSize(new_base_filename, NULL) + server.aof_last_incr_size;
-            server.aof_rewrite_base_size = server.aof_current_size;
+            server.aof_current_size = base_size + server.aof_last_incr_size;
+            server.aof_rewrite_base_size = base_size;
```

Comment:
@yoav-steinberg do you see any reason to include the new incremental file (previously the size of the AOFRW buffer) in the base size?

Reply:
One problem i can think is that if AOFRW is executed for a long time, and the amount of writing during this period is large, AOFRW may be triggered at a very fast frequency, but I think this is also the original intention of AOFRW.

Reply:
@yoav-steinberg  As I said above, the problem you mentioned does exist. A compromise method is that we continue to maintain the meaning of `aof_rewrite_base_size` when redis is running, and only set `aof_rewrite_base_size` to the size of BASE AOF when redis startup, which can not only solve the problems we encountered, but also avoid high-frequency AOFRW. 

I don't think it's worth modifying the manifest to get the exact `aof_rewrite_base_size`, because I don't think `aof_rewrite_base_size` is part of the manifest record. On the contrary, in the worst case, we will execute AOFRW at most once immediately after redis startsup, which I think is no problem, after all, this is a small probability event.

Reply:
ok, so i understand we all agree that in theory we need `aof_rewrite_base_size` to include part of the incremental file.
we also agree that the most correct thing would be not to lose this info on restart.
but we argue how important is it not to lose it on restart, and if it worth the effort of adding more metadata to the manifest.

considering the impact of such a thing is one excessive rewrite after a restart, i'm willing to overlook it for now, as long as we properly document in the code what's the right thing to do, why is it, and what compromises we made....
WDYT?

Reply:
Agree.

Reply:
Agree. When setting `aof_rewrite_base_size` on startup we can add a comment that this is a rough estimation because it's missing the incr file's size when we last created that base.